packages/editor/test/services/container.service.test.ts

```diff
@@ -0,0 +1,719 @@
+    expect(properties.length).toEqual(0);
+  });
+
+  test('object with object property', () => {
+    const schema = {
+      type: 'object',
+      properties: {
+        foo: {
+          type: 'object',
+          properties: {
+            bar: {type: 'string'}
+          }
+        }
+      }
+    };
+
+    const properties = getContainerProperties(schema, schema);
+    expect(properties.length).toEqual(0);
+  });
+
+  test('support multiple same $ref', () => {
+    const schema: JsonSchema = {
+      definitions: {
+        person: {
+          id: '#id',
+          type: 'object',
+          properties: {
+            name: {type: 'string'}
+          }
+        }
+      },
+      type: 'object',
+      properties: {
+        friends: {type: 'array', items: {$ref: '#/definitions/person'}},
+        enemies: {type: 'array', items: {$ref: '#/definitions/person'}}
+      }
+    };
+
+    const properties = getContainerProperties(schema, schema);
+    expect(properties.length).toEqual(2);
+    expect(getPropertyLabel(properties[0])).toEqual('person');
+    expect(getPropertyLabel(properties[1])).toEqual('person');
+    const personCopy = JSON.parse(JSON.stringify(schema.definitions.person));
+    expect(properties[0].schema).toMatchObject(personCopy);
+    expect(properties[1].schema).toMatchObject(personCopy);
+  });
+
+  test('support multiple different $ref', () => {
+    const schema: JsonSchema = {
+      definitions: {
+        person: {
+          type: 'object',
+          properties: {
+            name: {type: 'string'}
+          }
+        },
+        robot: {
+          type: 'object',
+          properties: {
+            id: {type: 'string'}
+          }
+        }
+      },
+      type: 'object',
+      properties: {
+        persons: {type: 'array', items: {$ref: '#/definitions/person'}},
+        robots: {type: 'array', items: {$ref: '#/definitions/robot'}}
+      }
+    };
+
+    const properties = getContainerProperties(schema, schema);
+    expect(properties.length).toEqual(2);
+    const personCopy = JSON.parse(JSON.stringify(schema.definitions.person));
+    personCopy.id = '#' + (schema.properties.persons.items as JsonSchema).$ref;
+    const robotCopy = JSON.parse(JSON.stringify(schema.definitions.robot));
+    robotCopy.id = '#' + (schema.properties.robots.items as JsonSchema).$ref;
+    expect(getPropertyLabel(properties[0])).toEqual('person');
+    expect(getPropertyLabel(properties[1])).toEqual('robot');
+    expect(properties[0].schema).toMatchObject(personCopy);
+    expect(properties[1].schema).toMatchObject(robotCopy);
+  });
+
+  test('support root anyOf', () => {
+    const schema: JsonSchema = {
+      definitions: {
+        a: {type: 'object'},
+        b: {type: 'object'}
+      },
+      anyOf: [
+        {$ref: '#/definitions/a'},
+        {$ref: '#/definitions/b'}
+      ]
+    };
+
+    const properties = getContainerProperties(schema, schema);
+    expect(properties.length).toEqual(0);
```

Comment:
Does this return 0 because the `a` and `b` definitions do not have `properties`?